library/catalog/models.py

```diff
@@ -45,6 +46,44 @@ class MongoUser(Document):
     rated_books = DictField(default={})
     preference = EmbeddedDocumentField(Preference.__name__, default=Preference())
 
+    def make_recommended_books_list(self, books_limit=50):
+        if self.wishlist:
+            wishlist_books = [Book.objects(id=book_id).first() for book_id in self.wishlist]
+            wishlist_books.sort(key=lambda book: book.statistic.rating, reverse=True)
+
+            books_by_author_and_genre = []
+            books_by_author = []
+            books_by_genre = []
+
+            for book in wishlist_books:
+                books_by_author_and_genre += list(Book.objects.filter(
+                    Q(author_id=book.author_id) & Q(genres__icontains=book.genres)).order_by('-statistic__rating'))
+                books_by_author += list(Book.objects.filter(author_id=book.author_id).order_by('-statistic__rating'))
+                books_by_genre += list(
+                    Book.objects.filter(genres__icontains=book.genres).order_by('-statistic__rating'))
+
+            recommendations_list = books_by_author_and_genre + books_by_author + books_by_genre
+            recommendations_list.reverse()
+            for book in recommendations_list:
+                while recommendations_list.count(book) != 1:
+                    recommendations_list.remove(book)
+            recommendations_list.reverse()
+
+            recommended_books = [str(book.id) for book in recommendations_list]
+            for book_id in self.wishlist:
+                if book_id in recommended_books:
+                    recommended_books.remove(book_id)
+            self.recommended_books = recommended_books[:books_limit]
+
+        else:
+            top_books = list(Book.objects().order_by('-statistic__rating'))[:books_limit // 2]
+            random_books = random.choices(list(set(Book.objects()) - set(top_books)), k=books_limit - books_limit // 2)
```

Comment:
You may just show top books from the rating.

Reply:
Yes, it's another way, but we have already top books on the 'home' page.
I think recommended mix from the top and random books for proposing something new